test/mainnet/curve.spec.sol

```diff
@@ -0,0 +1,213 @@
+            targets[0] = address(pool);
+
+            uint256[] memory values = new uint256[](1);
+            values[0] = swapAmount; // Send ETH with the call
+
+            bytes[] memory callData = new bytes[](1);
+            callData[0] = exchangeData;
+
+            vm.startPrank(PROCESSOR);
+            vault.processor(targets, values, callData);
+            vm.stopPrank();
+        }
+
+        // Assert ETH balance is reduced and stETH balance matches expected amount
+        assertApproxEqAbs(address(vault).balance, amount - swapAmount, 2);
+        assertApproxEqAbs(IERC20(MC.STETH).balanceOf(address(vault)), minOut, 2);
+    }
+
+    function test_Vault_Curve_addLiquidity() public {
+        uint256 ethAmount = 100 ether;
+        uint256 stethAmount = 100 ether;
+
+        // User deposits ETH
+        address user = makeAddr("user");
+        vm.deal(user, ethAmount + stethAmount);
+        vm.startPrank(user);
+        (bool success,) = address(vault).call{value: ethAmount}("");
+        require(success, "ETH transfer failed");
+
+        // Get stETH by depositing ETH to Lido
+        (success,) = MC.STETH.call{value: stethAmount}("");
+        require(success, "stETH deposit failed");
+        IERC20(MC.STETH).approve(address(vault), stethAmount);
+        IERC20(MC.STETH).transfer(address(vault), stethAmount);
+        vm.stopPrank();
+
+        // Get curve pool for ETH/stETH
+        ICurveRegistry registry = ICurveRegistry(MC.CURVE_REGISTRY);
+        ICurvePool pool = ICurvePool(registry.find_pool_for_coins(MC.ETH, MC.STETH));
+
+        // Prepare amounts array for add_liquidity
+        uint256[2] memory amounts = [ethAmount, stethAmount];
+        uint256 minLpOut = (pool.calc_token_amount(amounts, true) * 999) / 1000; // 0.1% slippage
+
+        // Prepare add_liquidity data
+        bytes memory addLiquidityData = abi.encodeWithSelector(
+            bytes4(keccak256("add_liquidity(uint256[2],uint256)")),
+            amounts,
+            minLpOut
+        );
+        {
+            address[] memory targets = new address[](2);
+            targets[0] = MC.STETH;
+            targets[1] = address(pool);
+
+            uint256[] memory values = new uint256[](2);
+            values[0] = 0;
+            values[1] = ethAmount; // Send ETH with the call
+
+            bytes[] memory callData = new bytes[](2);
+            callData[0] = abi.encodeWithSignature("approve(address,uint256)", address(pool), stethAmount);
+            callData[1] = addLiquidityData;
+
+            vm.startPrank(PROCESSOR);
+            vault.processor(targets, values, callData);
+            vm.stopPrank();
+        }
+
+        // Assert ETH balance is reduced and LP tokens were received
+        assertApproxEqAbs(address(vault).balance, 0, 2);
+        assertApproxEqAbs(IERC20(MC.STETH).balanceOf(address(vault)), 0, 2);
+        uint256 lpBalance = IERC20(pool.lp_token()).balanceOf(address(vault));
+        assertGt(lpBalance, minLpOut);
```

Comment:
important note here: if we support add_liquidity then we need the LP pool as a strategy and a rate for it

Reply:
yeah, let's sync on this arch topic today